firebase-web/src/test/java/io/spine/web/firebase/rest/HttpClientMockFactory.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.web.firebase.rest;
+
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.LowLevelHttpRequest;
+import com.google.api.client.testing.http.MockHttpTransport;
+import com.google.api.client.testing.http.MockLowLevelHttpRequest;
+import com.google.api.client.testing.http.MockLowLevelHttpResponse;
+
+import java.io.IOException;
+
+/**
+ * Creates mocked {@link HttpClient}s.
+ */
+final class HttpClientMockFactory {
+
+    /**
+     * Prevents instantiation of this utility class.
+     */
+    private HttpClientMockFactory() {
+    }
+
+    /**
+     * Creates a new no-operation {@code HttpClient}.
+     */
+    static HttpClient noOpClient() {
+        MockHttpTransport transport = new MockHttpTransport.Builder().build();
+        return HttpClient.using(transport.createRequestFactory());
+    }
+
+    /**
+     * Creates an {@code HttpClient} mock which returns the specified {@code content}
+     * on every request.
+     */
+    static HttpClient mockHttpClient(String content) {
+        return mockHttpClient(content, HttpClientMockFactory::noOpObserver);
+    }
+
+    /**
+     * Creates an {@code HttpClient} mock which returns the specified {@code content}
+     * on every request and uses supplied {@code observer} while building requests.
+     */
+    static HttpClient mockHttpClient(String content, RequestObserver observer) {
```

Comment:
I don't understand neither the naming nor the purpose of `RequestObserver`. Could you please elaborate on this so that this API piece looked a bit more clear to a naked eye?